internal/socket/rawaddr.go

```diff
@@ -0,0 +1,41 @@
+package socket
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+)
+
+// todo: should these be custom types to store the desired/actual size and addr family?
+
+var (
+	ErrBufferSize     = errors.New("buffer size")
+	ErrInvalidPointer = errors.New("invalid pointer")
+	ErrAddrFamily     = errors.New("address family")
+)
+
+// todo: replace this with generics, along with GetSockName and co.
```

Comment:
If you plan on taking these up at some point, probably best to `TODO: helsaawy - ...` for todos